keymutex/hashed.go

```diff
@@ -42,17 +42,17 @@ type hashedKeyMutex struct {
 
 // Acquires a lock associated with the specified ID.
 func (km *hashedKeyMutex) LockKey(id string) {
-	km.mutexes[km.hash(id)%len(km.mutexes)].Lock()
+	km.mutexes[km.hash(id)%uint(len(km.mutexes))].Lock()
 }
 
 // Releases the lock associated with the specified ID.
 func (km *hashedKeyMutex) UnlockKey(id string) error {
-	km.mutexes[km.hash(id)%len(km.mutexes)].Unlock()
+	km.mutexes[km.hash(id)%uint(len(km.mutexes))].Unlock()
```

Comment:
I also think it's fine to cast the return of `len()` (`int`) into `uint` since `len` is always returning a zero or positive value.